File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015VB/VirtualRadGridViewSortingFiltering.Core/VirtualGroupBuilder.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls.Data
Imports Telerik.WinControls.UI
Imports System.Threading.Tasks

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class VirtualGroupBuilder
        Inherits GroupBuilder(Of GridViewRowInfo)
        Private dataView As VirtualGridDataView

        Public Sub New(index As Telerik.Collections.Generic.Index(Of GridViewRowInfo), dataView As VirtualGridDataView)
            MyBase.New(index)
            Me.dataView = dataView
        End Sub

        Protected Overrides Function GetItemKey(item As GridViewRowInfo, descriptor As SortDescriptor) As Object
            Return Me.dataView.ItemsSource.GetValue(item.DataBoundItem, descriptor.PropertyIndex)
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: GridView/VirtualRadGridView/VirtualRadGridViewCS/VirtualRadGridViewSortingFiltering.Core/VirtualGroupBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.Data;
using Telerik.WinControls.UI;
using System.Threading.Tasks;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class VirtualGroupBuilder : GroupBuilder<GridViewRowInfo>
    {
        private VirtualGridDataView dataView;

        public VirtualGroupBuilder(Telerik.Collections.Generic.Index<GridViewRowInfo> index, VirtualGridDataView dataView)
            : base(index)
        {
            this.dataView = dataView;
        }

        protected override object GetItemKey(GridViewRowInfo item, SortDescriptor descriptor)
        {
            return this.dataView.ItemsSource.GetValue(item.DataBoundItem, descriptor.PropertyIndex);
        }
    }
}
